ibu-imager/seedcreator/seedcreator.go

```diff
@@ -500,19 +496,37 @@ func (s *SeedCreator) waitTillOvnKubeNodeIsDown(ctx context.Context) {
 	defer deadlineCancel()
 }
 
-func (s *SeedCreator) getCatalogImages(ctx context.Context) ([]string, error) {
+// filterCatalogImages filters catalog source images as catalog sources have pull always policy
+// and there is no point to precache them.
+// List of catalog images are taken from catalog sources that are currently configured on cluster and known ocp ones.
+// Filtering known images will allow us to fix the case where those images were pulled in seed
+// and after it their catalog sources were removed but images are still part of podman images output as we create seed
+func (s *SeedCreator) filterCatalogImages(ctx context.Context, images []string) ([]string, error) {
+	// Regex matches list of known catalog images, there are 4 of them at least in 4.15
+	//registry.redhat.io/redhat/community-operator-index:v4.15
+	//registry.redhat.io/redhat/redhat-operator-index:v4.15
+	//registry.redhat.io/redhat/certified-operator-index:v4.15
+	//registry.redhat.io/redhat/redhat-marketplace-index:v4.15
+	defaultCatalogsRegex := regexp.MustCompile(`^registry\.redhat\.io\/redhat\/.+?-index:.+?`)
```

Comment:
Why the question marks?

Also you don't have to escape `\/`, `/` works too

Reply:
The last question mark seems to be wrong:

![image](https://github.com/openshift-kni/lifecycle-agent/assets/10882062/79fcca7e-123e-4127-823b-8360464c4b3a)